The null ArtifactsPath value causes exception for all EtwProfiler

wojtpl2 opened the pull request:
Some improvements for #1235. 
This is just nicer error handling. Unfortunately, this will not correct the mistake itself.

wojtpl2 commented:
The last commit resolve problem from #1235.
@adamsitnik It is ready for review and merge 😉

adamsitnik commented on src/BenchmarkDotNet/Configs/ImmutableConfigBuilder.cs:
```diff
@@ -61,7 +61,7 @@ public static ImmutableConfig Create(IConfig source)
                 uniqueRunnableJobs,
                 source.UnionRule,
-                source.ArtifactsPath,
+                source.ArtifactsPath ?? DefaultConfig.Instance.ArtifactsPath,
```
@wojtpl2 could you please add a test to `ImmutableConfigTests` that ensures that `ImmutableConfig.ArtifactsPath` is never null? I just want to make sure we don't regress in the future.

wojtpl2 commented:
@adamsitnik I've added tests for empty ArtifactsPath and Orderer. 

Sorry that I've removed trailing whitespaces. If you want, I can undo these changes.
I have https://marketplace.visualstudio.com/items?itemName=MadsKristensen.TrailingWhitespaceVisualizer

adamsitnik commented:
> Sorry that I've removed trailing whitespaces

No worries, GH allows me to hide the whitespace changes:

![obraz](https://user-images.githubusercontent.com/6011991/64112342-6a58e300-cd87-11e9-9780-e760ffbe5717.png)
